overlore/graphql/event.py

```diff
@@ -53,19 +57,48 @@ async def torii_boot_sync(torii_service_endpoint: str) -> None:
 
 
 async def torii_event_sub(
-    torii_service_endpoint: str, on_event_callback: OnEventCallbackType, subscription: Subscriptions
+    session: Client, on_event_callback: OnEventCallbackType, gql_subscription: Subscriptions
 ) -> None:
-    transport = WebsocketsTransport(url=torii_service_endpoint)
+    logger.debug("subscribing to %s", gql_subscription)
+    async for result in session.subscribe(gql(gql_subscription.value)):  # type: ignore[attr-defined]
+        try:
+            on_event_callback(result)
+        except RuntimeError:
+            logger.error("Unable to process event %s in %s", result, gql_subscription)
+
+
+def backoff_logging(details: Details) -> None:
+    # Retrieve exception type, value, and traceback
+    exc_type, exc_value, _ = sys.exc_info()
+
+    # Format the exception information for logging
+    if exc_type is not None and exc_value is not None:
+        exc_info = f"{exc_type.__name__}: {exc_value}"
+    else:
+        exc_info = "No exception information"
+
+    # Format the exception information for logging
+    logger.warning(
+        "Backing off %s seconds after %s tries calling function %s with args %s and kwargs %s due to %s",
+        details["wait"],
+        details["tries"],
+        details["target"].__name__,
+        details["args"],
+        details["kwargs"],
+        exc_info,
+    )
+
 
+@backoff.on_exception(backoff.expo, Exception, max_time=300, on_backoff=backoff_logging)
```

Comment:
Is there a reason for 5 minutes max_time? Shouldn't we just try indefinitely?

Reply:
Agreed about constraining the catching of exception to only networking exception